sky/resources.py

```diff
@@ -168,6 +167,8 @@ def __init__(
                 }
         self._is_image_managed = _is_image_managed
 
+        if isinstance(disk_tier, str):
+            disk_tier = resources_utils.DiskTier(str(disk_tier).lower())
```

Comment:
Should we update `__setstate__` for backward compatibility if a previous cluster was stored in the local db with a `str` disk tier?